### ISSUE TITLE
The matching doesn't seem ideal...

### ISSUE THREAD

**Issue by lovettchris** [DynamicTimeWarping/dtw-python#59]
**Describe the bug**
The matching in the following doesn't seem ideal, I have circled in red something that I believe should have matched up like a "ladder" with horizontal lines connecting the red and blue dots.

[compare_paths.zip](https://github.com/DynamicTimeWarping/dtw-python/files/11549868/compare_paths.zip)

**To Reproduce**
Open the attached zip and run the compare_paths jupyter notebook.

![image](https://github.com/DynamicTimeWarping/dtw-python/assets/18707114/2fab48d0-ce54-4932-9767-93dcdf0a1b56)

**Version**

dtw-python                1.3.0

**Comment by tonigi** [comment]
It's not a bug. The matching is "optimal" in the mathematical sense. Consider that it is based on the Euclidean distance between 3-vectors whose components have very different magnitudes. You may try to attune it to your intuition by e.g.: standardizing the timeseries' components, or changing the distance metric.

**Comment by lovettchris** [comment]
Thanks for looking at the bug, can you explain in layman's terms why it doesn't turn the circled section into a ladder of horizontal lines? Surely those vectors would be much shorter?

**Comment by tonigi** [comment]
It's indeed surprising as the "y" component should guide the alignment (it's the largest). I'm having a look but i still haven't found anything obviously wrong.

**Comment by lovettchris** [comment]
Oh, never mind, it was a stupid bug in my rendering, it should be this:

![image](https://github.com/DynamicTimeWarping/dtw-python/assets/18707114/7912dcde-f86d-46b3-a337-042f848d9ed2)

I was clobbering my "i" variable (dang python):
```python
    for index in range(len(index1)):
        i = int(index1[index])
        j = int(index2[index])
        x1, y1, z1 = path1[i]
        x2, y2, z2 = path2[j]
        xs += [(x1, x2)]
        ys += [(y1, y2)]
```

**Comment by tonigi** [comment]
Thanks! I suspected that but could not see the problem.

**Comment by lovettchris** [comment]
Sorry for wasting your time, and thanks for the response.
